addon/lib/metrics.js

```diff
@@ -62,93 +57,81 @@ const variantMaker = {
 };
 
 
-const Metrics = module.exports = {
+function makeTimestamp(time) {
+  const timestamp = typeof time !== 'undefined' ?  time : Date.now();
+  return Math.round((timestamp - store.browserLoadedTimestamp) / 1000);
+}
 
-  init: function() {
-    if (!store.telemetryPingPayload) {
-      store.lastTelemetryPingTimestamp = false;
-      store.telemetryPingPayload = {
-        version: 1,
-        tests: {}
-      };
-    }
 
-    Metrics.maybePingTelemetry();
+const Metrics = module.exports = {
 
+  init: function() {
+    store.browserLoadedTimestamp = Date.now();
     Events.on(EVENT_SEND_METRIC, Metrics.onExperimentPing);
     Events.on(EVENT_RECEIVE_VARIANT_DEFS, Metrics.onReceiveVariantDefs);
   },
 
   onEnable: function() {
-    // Backup existing preference settings and then override.
-    store.metricsPrefsBackup = {};
-    Object.keys(PREFERENCE_OVERRIDES).forEach(name => {
-      store.metricsPrefsBackup[name] = PrefsService.get(name);
-      PrefsService.set(name, PREFERENCE_OVERRIDES[name]);
-    });
+    Metrics.pingTelemetry(self.id, 'enabled', Date.now());
+    Metrics.prefs.backup();
   },
 
   onDisable: function() {
-    // Restore previous preference settings before override.
-    if (store.metricsPrefsBackup) {
+    Metrics.pingTelemetry(self.id, 'disabled', Date.now());
+    Metrics.prefs.restore();
+  },
+
+  prefs: {
```

Comment:
Are either of these `Metrics.prefs.{backup,restore}` methods called outside this module? If so, they could just be module-private functions like `makeTimestamp`